src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSBucketCreate.java

```diff
@@ -221,7 +245,136 @@ void validate(FailureCollector collector) {
             collector.addFailure(e.getMessage(), null).withConfigElement(NAME_PATHS, path);
           }
         }
-        collector.getOrThrowException();
+      }
+
+      if (!containsMacro(NAME_CMEK_KEY) && !Strings.isNullOrEmpty(cmekKey)) {
+        validateCmekKey(collector);
+      }
+      collector.getOrThrowException();
+    }
+
+    //This method validated the pattern of CMEK Key resource ID.
+    void validateCmekKey(FailureCollector failureCollector) {
```

Comment:
there is a lot of duplicate code. Please refactor so it is in one place.

Reply:
Refractored the code and moved that into functions in base class